src/sparse_blas/common_op_verification.hpp

```diff
@@ -0,0 +1,142 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef _ONEMKL_SRC_SPARSE_BLAS_COMMON_OP_VERIFICATION_HPP_
+#define _ONEMKL_SRC_SPARSE_BLAS_COMMON_OP_VERIFICATION_HPP_
+
+#include <string>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/sparse_blas/types.hpp"
+#include "macros.hpp"
+
+namespace oneapi::mkl::sparse::detail {
+
+/// Throw an exception if the scalar is not accessible in the host
+inline void check_ptr_is_host_accessible(const std::string &function_name,
+                                         const std::string &scalar_name,
+                                         bool is_ptr_accessible_on_host) {
+    if (!is_ptr_accessible_on_host) {
+        throw mkl::invalid_argument(
+            "sparse_blas", function_name,
+            "Scalar " + scalar_name + " must be accessible on the host for buffer functions.");
+    }
+}
+
+template <typename InternalSparseMatHandleT>
+void check_valid_spmm_common(const std::string &function_name,
+                             oneapi::mkl::sparse::matrix_view A_view,
+                             InternalSparseMatHandleT internal_A_handle,
+                             oneapi::mkl::sparse::dense_matrix_handle_t B_handle,
+                             oneapi::mkl::sparse::dense_matrix_handle_t C_handle,
+                             bool is_alpha_host_accessible, bool is_beta_host_accessible) {
+    THROW_IF_NULLPTR(function_name, internal_A_handle);
+    THROW_IF_NULLPTR(function_name, B_handle);
+    THROW_IF_NULLPTR(function_name, C_handle);
+
+    check_all_containers_compatible(function_name, internal_A_handle, B_handle, C_handle);
+    if (internal_A_handle->all_use_buffer()) {
+        check_ptr_is_host_accessible("spmm", "alpha", is_alpha_host_accessible);
+        check_ptr_is_host_accessible("spmm", "beta", is_beta_host_accessible);
+    }
```

Comment:
Is this check for USM or `sycl::buffer`? Looks like it should be for USM, but it is being done for buffer? Maybe needs to be `if !(internal_A_handle->all_use_buffer())`

Reply:
Ah, nevermind. These checks really are needed for `sycl::buffer` as _that's_ the one that doesn't allow device pointers for `alpha`/`beta`.